base_layer/core/src/transactions/transaction_components/transaction_output.rs

```diff
@@ -476,3 +481,92 @@ impl ConsensusDecoding for TransactionOutput {
         Ok(output)
     }
 }
+
+pub fn batch_verify_range_proofs(
+    prover: &RangeProofService,
+    outputs: &[&TransactionOutput],
+) -> Result<(), RangeProofError> {
+    // Batched range proof verification gains above batch sizes of 256 gives diminishing returns,
+    // see <https://github.com/tari-project/bulletproofs-plus>
+    // Example: If we have 15 outputs, then we need chunks of 8, 4, 2, 1.
+    let power_of_two_vec = power_of_two_chunk_sizes(outputs.len(), 8);
+    debug!(
+        target: LOG_TARGET,
+        "Queueing range proof batch verify output(s): {:?}", &power_of_two_vec
+    );
+    let mut index = 0;
+    for power_of_two in power_of_two_vec {
+        let mut statements = Vec::with_capacity(power_of_two);
+        let mut proofs = Vec::with_capacity(power_of_two);
+        for output in outputs.iter().skip(index).take(power_of_two) {
+            statements.push(RistrettoAggregatedPublicStatement {
+                statements: vec![Statement {
+                    commitment: output.commitment.clone(),
+                    minimum_value_promise: 0,
+                }],
+            });
+            proofs.push(output.proof.to_vec().clone());
+        }
+        index += power_of_two;
+        prover.verify_batch(proofs.iter().collect(), statements.iter().collect())?;
+    }
+    Ok(())
+}
+
```

Comment:
This is a pub function. Does this need to be public?
Either way, this function can benefit from docs. To understand what this function does, you need to read the comment on the function `batch_verify_range_proofs` and the tests for this function, this is not ideal.

Reply:
Good point, fixed